testsuite/test_wbemconnection_mock.py

```diff
@@ -4367,3 +4372,166 @@ def test_invokemethod(self, conn, ns, desc, inputs, exp_output,
+        shutil.rmtree(schema_dir)
+
+
+class TestDMTFSchema(object):
+    """
+    Test the DMTFSchema class in pywbem_mock.  Since we do not want to always
+    be downloading the DMTF schema for CI testing, we will generally skip the
+    first test.
+    """
+
+    def test_schema_final_load(self, test_schema_dir):
+        # pylint: disable=no-self-use
+        """
+        Test the DMTFSchema class and its methods to get a schema from the
+        DMTF, expand it, and to create a partial mof schema definition for
+        compilation.
+        """
+        schema = DMTFSchema((2, 49, 0), test_schema_dir, verbose=True)
+
+        assert schema.schema_ver == (2, 49, 0)
+        assert schema.schema_ver_str == '2.49.0'
+        assert os.path.isdir(schema.schema_dir)
+        assert os.path.isdir(schema.schema_mof_dir)
+        assert os.path.isfile(schema.schema_mof_filename)
+        assert schema.schema_dir == test_schema_dir
+        mof_dir = 'mofFinal%s' % schema.schema_ver_str
+        test_schema_mof_dir = os.path.join(test_schema_dir, mof_dir)
+        assert schema.schema_mof_dir == test_schema_mof_dir
+
+        assert schema.schema_mof_filename == \
+            os.path.join(test_schema_mof_dir,
+                         'cim_schema_%s.mof' % (schema.schema_ver_str))
+
+        assert repr(schema) == "DMTFSchema(schema_ver=(2, 49, 0), " \
+                               "schema_dir=%s)" % test_schema_dir
+
+        schema_mof = schema.build_schema_mof(['CIM_ComputerSystem', 'CIM_Door'])
+
+        exp_mof = '#pragma locale ("en_US")\n' \
+                  '#pragma include ("System/CIM_ComputerSystem.mof")\n' \
+                  '#pragma include ("Device/CIM_Door.mof")\n'
+
+        assert schema_mof == exp_mof
+
+    def test_schema_experimental_load(self, test_schema_dir):
+        # pylint: disable=no-self-use
+        """
+        Test the DMTFSchema class and its methods to get a schema from the
+        DMTF, expand it, and to create a partial mof schema definition for
+        compilation.
+        """
+        schema = DMTFSchema((2, 49, 0), test_schema_dir, use_experimental=True,
+                            verbose=True)
+
+        assert schema.schema_ver == (2, 49, 0)
+        assert schema.schema_ver_str == '2.49.0'
+        assert os.path.isdir(schema.schema_dir)
+        assert os.path.isdir(schema.schema_mof_dir)
+        assert os.path.isfile(schema.schema_mof_filename)
+        assert schema.schema_dir == test_schema_dir
+        mof_dir = 'mofExperimental%s' % schema.schema_ver_str
+        test_schema_mof_dir = os.path.join(test_schema_dir, mof_dir)
+        assert schema.schema_mof_dir == test_schema_mof_dir
+
+        assert schema.schema_mof_filename == \
+            os.path.join(test_schema_mof_dir,
+                         'cim_schema_%s.mof' % (schema.schema_ver_str))
+
+    def test_two_schema_load(self, test_schema_dir):
+
+        schema1 = DMTFSchema((2, 49, 0), test_schema_dir, verbose=True)
```

Comment:
We should use the version from the single place where we define it, so that it is ensured that that version is the one we persist.

Reply:
All tied back to the original definition now.

NOTE: I still am downloading schema for some the tests and will change that as the last change.

DONE